### PR TITLE
Report fail if binding to a socket fails in websockets

### PR DESCRIPTION
**Pull request by Symbiatch**
Fixes issue #2533 when websocket server cannot be created due to a bind failure

### CODE REVIEW

**Comment by januscla** [comment]
Thanks for your contribution, @Symbiatch! Please make sure you sign our [CLA](https://cla.conf.meetecho.com/janus), as it's a required step before we can merge this.

File: transports/janus_websockets.c [hunk 1]
```diff
@@ -636,7 +636,11 @@ int janus_websockets_init(janus_transport_callbacks *callback, const char *confi
 			info.ssl_private_key_password = NULL;
 			info.gid = -1;
 			info.uid = -1;
-			info.options = 0;
+#if (LWS_LIBRARY_VERSION_MAJOR == 3 && LWS_LIBRARY_VERSION_MINOR >= 2) || (LWS_LIBRARY_VERSION_MAJOR > 3)
+                        info.options = LWS_SERVER_OPTION_FAIL_UPON_UNABLE_TO_BIND;
+#else
+                        info.options = 0;
+#endif
```
Comment (lminiero) [inline transports/janus_websockets.c:643]: Indentation seems broken? Please use tabs, not spaces (code style nit).
As a side note, no need for the `else` part with the `0` init: we already memset the whole `info` object to `0` when we declare the variable.

File: transports/janus_websockets.c [hunk 2]
```diff
@@ -753,7 +759,11 @@ int janus_websockets_init(janus_transport_callbacks *callback, const char *confi
...
```
Comment (lminiero) [inline transports/janus_websockets.c:766]: Same here.

File: transports/janus_websockets.c [hunk 3]
```diff
@@ -818,7 +828,9 @@ int janus_websockets_init(janus_transport_callbacks *callback, const char *confi
 				info.ssl_cipher_list = ciphers;
 				info.gid = -1;
 				info.uid = -1;
-#if LWS_LIBRARY_VERSION_MAJOR >= 2
+#if (LWS_LIBRARY_VERSION_MAJOR == 3 && LWS_LIBRARY_VERSION_MINOR >= 2) || (LWS_LIBRARY_VERSION_MAJOR > 3)
+                                info.options =  LWS_SERVER_OPTION_DO_SSL_GLOBAL_INIT | LWS_SERVER_OPTION_FAIL_UPON_UNABLE_TO_BIND;
+#elif LWS_LIBRARY_VERSION_MAJOR >= 2
```
Comment (lminiero) [inline transports/janus_websockets.c:833]: Same here. There's also an extra space after the equal sign that shouldn't be there.

**Review by lminiero — COMMENTED** [review]
Thanks for taking care of this! I'll make a couple of tests later to check if it works as expected in my setup. I added a couple of notes inline.

**Comment by Symbiatch** [comment]
> 
> 
> Thanks for taking care of this! I'll make a couple of tests later to check if it works as expected in my setup. I added a couple of notes inline.

Thanks, somehow the indentation was mangled, I'll fix it and remove the zeroes, they were there originally so kept them to be sure.

**Comment by lminiero** [comment]
> they were there originally so kept them to be sure

Ah you're right, the diff had confused me! Removing them is still fine, since the memset does indeed take care of it.
Thanks for the quick patch! I'll test and get back to you later.

**Comment by atoppi** [comment]
lgtm :+1:

**Comment by lminiero** [comment]
LGTM too, thanks, merging! :+1:

**Pull request merged**